### PR TITLE
drive: Added rc support for changing the service-account-file

### PR DESCRIPTION
**Pull request by darthShadow**
Signed-off-by: Anagh Kumar Baranwal <6824881+darthShadow@users.noreply.github.com>

#### What is the purpose of this change?

Add rc support for changing the service-account-file

#### Was the change discussed in an issue or in the forum before?

https://forum.rclone.org/t/edit-service-account-while-mounted/15760

#### Checklist

- [X] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-pull-request).
- [ ] I have added tests for all changes in this PR if appropriate.
- [ ] I have added documentation for the changes if appropriate.
- [ ] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [ ] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

File: backend/drive/drive.go [hunk 1]
```diff
@@ -542,6 +543,8 @@ type Fs struct {
 	exportExtensions []string           // preferred extensions to download docs
 	importMimeTypes  []string           // MIME types to convert to docs
 	isTeamDrive      bool               // true if this is a team drive
+
+	configMapper     configmap.Mapper   // TODO: Figure a way around this
```
Comment (darthShadow) [inline backend/drive/drive.go:547]: Any way to avoid this and still be able to reference the mapper for creating the oAuthClient?

File: fs/fs.go [hunk 2]
```diff
@@ -1286,6 +1286,12 @@ func NewFs(path string) (Fs, error) {
 	if err != nil {
 		return nil, err
 	}
+	// TODO: Fix import for rc
+	// f, err := fsInfo.NewFs(configName, fsPath, config)
+	// if err != nil && f.reload != nil {
+	// 	rc.AddOptionReload(f.name, &f.opt, f.reload)
+	// }
+	// return f, err
```
Comment (darthShadow) [inline fs/fs.go:1294]: I was planning to add it as a common option, but there is a cyclic import for the rc & fs libs. Thoughts on avoiding it or should I remove this for now?

**Comment by darthShadow** [comment]
@ncw This is for a dynamic reload of the service account via rc. I have 2 problems right now and have added comments inline for both of them. Would you mind taking a look?

**Comment by ncw** [comment]
I think this is a great idea, however I think doing it via the rc is the wrong framework... What if there is more than one drive backend in use?

This needs to in via the backend tools framework (which I have designed but not built yet!) so that we can specify a drive backend instantiation on the command line.

So I'd suggest trying to make it work with the rc as a prototype, then when I've merged the backend tools framework you can plug it into that which will give an `rc` command to access it (and also a command line tool which will be pretty useless in this case!).

**Comment by darthShadow** [comment]
I did think about the multiple drive backends, which is why the reload command is added per fs with the config remote name as the key. However, I have not tested that scenario myself so I assume you are right and I missed something which will stop it from working.
The only case that would break the above scenario is when the same config remote is specified multiple times and I am not sure how to go about handling that.

I agree, however, that this is not ideal and look forward to the proper way of integrating this.

So, how do we go further? Do you want me to fix the quality issue and get it ready for merging till the backend tools framework is brought in or close the PR for now?

**Comment by darthShadow** [comment]
Closing in order to handle this properly via #4180

**Pull request closed without merging**